tests/util/cluster_registry_test.go

```diff
@@ -0,0 +1,288 @@
+				Annotations: annotations,
+			},
+			Spec: k8s_cr.ClusterSpec{
+				KubernetesAPIEndpoints: k8s_cr.KubernetesAPIEndpoints{
+					ServerEndpoints: []k8s_cr.ServerAddressByClientCIDR{
+						{
+							ClientCIDR:    "10.23." + strconv.Itoa(i) + ".0/28",
+							ServerAddress: config.Clusters[0].Cluster.Server,
+						},
+					},
+				},
+			},
+		}
+		clusters[clusterName] = cluster
+	}
+
+	configmap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "clusterregistry",
+			Namespace: namespace,
+		},
+		Data: map[string]string{},
+	}
+	for cn, cd := range clusters {
+
+		dataBytes, err1 := json.Marshal(cd)
+		if err1 != nil {
+			return err1
+		}
+		configmap.Data[cn] = string(dataBytes)
+	}
+	_, err = client.CoreV1().ConfigMaps(namespace).Create(&configmap)
+
+	return err
+}
+
+func initMulticlusterPilot(IstioSrc string) (*bootstrap.Server, error) {
+
+	serverAgrs := bootstrap.PilotArgs{
+		Namespace: "istio-system",
+		DiscoveryOptions: envoy.DiscoveryServiceOptions{
+			Port:            18080, // An unused port will be chosen
+			GrpcAddr:        ":0",
+			EnableCaching:   true,
+			EnableProfiling: true,
+		},
+		//TODO: start mixer first, get its address
+		Mesh: bootstrap.MeshArgs{
+			MixerAddress:    "istio-mixer.istio-system:9091",
+			RdsRefreshDelay: ptypes.DurationProto(10 * time.Millisecond),
+		},
+		Config: bootstrap.ConfigArgs{
+			KubeConfig:                 IstioSrc + "/.circleci/mc-config",
+			ClusterRegistriesConfigmap: "clusterregistry",
+			ClusterRegistriesNamespace: "istio-testing",
+		},
+		Service: bootstrap.ServiceArgs{
+			Registries: []string{
+				string(serviceregistry.KubernetesRegistry)},
+		},
+	}
+	// Create the server for the discovery service.
+	discoveryServer, err := bootstrap.NewServer(serverAgrs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create discovery service: %v", err)
+	}
+
+	return discoveryServer, nil
+}
+
+func startMulticlusterPilot(s *bootstrap.Server, stop chan struct{}) {
+	// Start the server
+	_, _ = s.Start(stop)
+}
+
+func prepareEnv(t *testing.T) error {
+	var err error
+
+	urls, err := getAPIsURLs()
+	if err != nil {
+		return err
+	}
+	if len(urls) == 0 {
+		return fmt.Errorf("NOURL")
+	}
+	t.Logf("Retrieved URLs : %+v", urls)
+
+	clusterConfigs, err := createClusterConfig(urls)
+	if err != nil {
+		return err
+	}
+	if err = checkLocalAPIServer(urls[0]); err != nil {
+		return fmt.Errorf("error in checkLocalAPIServer: %v", err)
+	}
+	if err = ensureTestNamespace(urls[0], "istio-testing"); err != nil {
+		return fmt.Errorf("error in checkLocalAPIServer: %v", err)
+	}
+	if err = createSecretFromClusterConfig(urls[0], "istio-testing", clusterConfigs); err != nil {
+		return fmt.Errorf("error in createSecretFromClusterConfig: %v", err)
+	}
+
+	if err = createClusterConfigmap(urls[0], "istio-testing", clusterConfigs); err != nil {
+		return fmt.Errorf("error in createCusterConfigmap: %v", err)
+	}
+
+	return err
+}
+
+func TestClusterRegistry(t *testing.T) {
```

Comment:
I don't think util is the right directory for this test. Can you move it next to cluster registry  ?

Reply:
moved both to tests/local since both of these tests needs to run in local environment.